### ISSUE TITLE
plottr fails to launch with pyqtgraph related error

### ISSUE THREAD

**Issue by erikawa-e** [toolsforexperiments/plottr#246]
I am trying to install plottr following 
https://github.com/toolsforexperiments/plottr.

After installation, when I type 
"plottr-inspectr"
I got the following error:
AttributeError: module 'pyqtgraph' has no attribute 'ColorBarItem'

Thanks in advance.

**Comment by jenshnielsen** [comment]
@erikawa-e Could you check the version of pyqtgraph that you have installed. You could do that with `pip list`

Where you should see a line like this 
```
pyqtgraph                         0.12.3

```

**Comment by erikawa-e** [comment]
Package                       Version
----------------------------- ----------
alabaster                     0.7.12
appdirs                       1.4.4
argcomplete                   1.12.3
argh                          0.26.2
argon2-cffi                   21.1.0
arrow                         1.2.1
asteval                       0.9.25
astroid                       2.9.0
async-generator               1.10
atomicwrites                  1.4.0
attrs                         21.2.0
autopep8                      1.6.0
azure-core                    1.21.1
azure-identity                1.7.1
Babel                         2.9.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
bcrypt                        3.2.0
binaryornot                   0.4.4
black                         21.12b0
bleach                        4.1.0
Bottleneck                    1.3.2
broadbean                     0.10.0
brotlipy                      0.7.0
cachetools                    4.2.4
certifi                       2021.10.8
cffi                          1.15.0
chardet                       4.0.0
charset-normalizer            2.0.9
click                         8.0.3
cloudpickle                   2.0.0
colorama                      0.4.4
contextlib2                   0.5.5
cookiecutter                  1.7.0
cryptography                  36.0.1
cycler                        0.11.0
dataclasses                   0.8
debugpy                       1.5.1
decorator                     5.1.0
defusedxml                    0.7.1
diff-match-patch              20200713
docutils                      0.17.1
entrypoints                   0.3
flake8                        4.0.1
future                        0.18.2
google-api-core               2.3.2
google-auth                   2.3.3
googleapis-common-protos      1.54.0
h5netcdf                      0.11.0
h5py                          2.10.0
hypothesis                    6.0.0
idna                          3.1
imagesize                     1.3.0
importlib-metadata            4.8.2
importlib-resources           5.4.0
inflection                    0.5.1
iniconfig                     1.1.1
intervaltree                  3.0.2
ipykernel                     6.6.0
ipython                       7.30.1
ipython-genutils              0.2.0
ipywidgets                    7.6.5
isort                         5.10.1
jedi                          0.18.1
jellyfish                     0.8.9
Jinja2                        3.0.3
jinja2-time                   0.2.0
jsonschema                    4.3.0
jupyter                       1.0.0
jupyter-client                7.1.0
jupyter-console               6.4.0
jupyter-core                  4.9.1
jupyterlab-pygments           0.1.2
jupyterlab-widgets            1.0.2
keyring                       23.4.0
kiwisolver                    1.3.2
lazy-object-proxy             1.7.1
lmfit                         1.0.3
lxml                          4.7.1
MarkupSafe                    2.0.1
matplotlib                    3.3.2
matplotlib-inline             0.1.3
mccabe                        0.6.1
mistune                       0.8.4
mkl-fft                       1.3.0
mkl-random                    1.2.0
mkl-service                   2.3.0
more-itertools                8.12.0
msal                          1.16.0
msal-extensions               0.3.1
mypy-extensions               0.4.3
nbclient                      0.5.9
nbconvert                     6.3.0
nbformat                      5.1.3
nest-asyncio                  1.5.4
notebook                      6.4.6
numexpr                       2.7.3
numpy                         1.19.2
numpydoc                      1.1.0
olefile                       0.46
opencensus                    0.8.0
opencensus-context            0.1.2
opencensus-ext-azure          1.1.0
packaging                     21.3
pandas                        1.3.5
pandocfilters                 1.5.0
paramiko                      2.8.1
parso                         0.8.3
pathspec                      0.9.0
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        8.4.0
pip                           21.3.1
platformdirs                  2.3.0
plottr                        0.8.1
pluggy                        1.0.0
portalocker                   2.3.2
poyo                          0.5.0
prometheus-client             0.12.0
prompt-toolkit                3.0.24
protobuf                      3.19.1
psutil                        5.8.0
ptyprocess                    0.7.0
py                            1.11.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycodestyle                   2.8.0
pycparser                     2.21
pydocstyle                    6.1.1
pyflakes                      2.4.0
Pygments                      2.10.0
PyJWT                         2.3.0
pylint                        2.12.2
pyls-spyder                   0.4.0
PyNaCl                        1.4.0
pyOpenSSL                     21.0.0
pyparsing                     3.0.6
PyQt5                         5.15.6
PyQt5-Qt5                     5.15.2
PyQt5-sip                     12.9.0
pyqtgraph                     0.11.0
pyreadline                    2.1
pyrsistent                    0.18.0
PySide2                       5.15.2
PySocks                       1.7.1
pytest                        6.2.5
pytest-runner                 5.3.1
python-dateutil               2.8.2
python-lsp-black              1.0.1
python-lsp-jsonrpc            1.0.0
python-lsp-server             1.3.3
python-slugify                5.0.2
pytz                          2021.3
PyVISA                        1.11.1
pywin32                       302
pywin32-ctypes                0.2.0
pywinpty                      1.1.6
PyYAML                        6.0
pyzmq                         22.3.0
qcodes                        0.30.1
QDarkStyle                    3.0.2
qstylizer                     0.2.1
QtAwesome                     1.1.1
qtconsole                     5.2.2
QtPy                          1.11.3
regex                         2021.11.10
requests                      2.26.0
requirements-parser           0.2.0
rope                          0.22.0
rsa                           4.8
Rtree                         0.9.7
ruamel.yaml                   0.17.17
ruamel.yaml.clib              0.2.6
schema                        0.7.5
scipy                         1.7.3
Send2Trash                    1.8.0
setuptools                    59.6.0
setuptools-scm                6.3.2
shiboken2                     5.15.2
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
Sphinx                        4.3.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.2.1
spyder-kernels                2.2.0
tabulate                      0.8.9
terminado                     0.12.1
testpath                      0.5.0
text-unidecode                1.3
textdistance                  4.2.2
three-merge                   0.1.1
tinycss                       0.4
tinycss2                      1.1.1
toml                          0.10.2
tomli                         1.2.2
tornado                       6.1
tqdm                          4.62.3
traitlets                     5.1.1
typed-ast                     1.5.1
typing_extensions             4.0.1
ujson                         4.2.0
uncertainties                 3.1.6
Unidecode                     1.3.2
urllib3                       1.26.7
watchdog                      2.1.6
wcwidth                       0.2.5
webencodings                  0.5.1
websockets                    8.1
wheel                         0.37.0
whichcraft                    0.6.1
widgetsnbextension            3.5.2
win-inet-pton                 1.1.0
wincertstore                  0.2
wrapt                         1.13.3
xarray                        0.20.2
yapf                          0.31.0
zipp                          3.6.0

**Comment by erikawa-e** [comment]
It looks like that we have a version of 0.11.0 for pyqtgraph.

**Comment by jenshnielsen** [comment]
Could you try upgrading it to 0.12.3 it may be that plottr requires 0.12 (but does not correctly state that as a requirement)

**Comment by jenshnielsen** [comment]
I can reproduce the issue with pyqtgraph 0.11.0 and 0.12.0 but not 0.12.1 so #247 limits the minimum version to that

**Comment by erikawa-e** [comment]
Thanks.

We installed the newer version with "pip install pyqtgraph==0.12.3".

Then, after typing 
"plottr-inspectr"
now I got the new error:
ValueError: PyCapsule_GetPointer called with incorrect name

**Comment by jenshnielsen** [comment]
Could you check the versions of pyqt5 that you have installed? And perhaps try to upgrade to the latest version

**Comment by erikawa-e** [comment]
The version of pyqt5 is 5.15.6 and I am afraid that it is the latest one...

**Comment by jenshnielsen** [comment]
Thanks not, Strange I am not really sure what is wrong beyond the fact that it seems related to pyqt 

Could you try creating a new environment and see if that has the same issue. Alternatively uninstall all the pyqt packages and reinstall them 

Are you by any chance mixing pyqt packages installed with pip and conda? 

See https://stackoverflow.com/questions/55794101/valueerrorpycapsule-getpointer-called-with-incorrect-name-with-from-pyqt5-qtwe

If your environment is created with conda could you paste the output of conda list too?

**Comment by erikawa-e** [comment]
OK! Thanks! I will try to create a new environment and come back to you with the result soon!

**Comment by jenshnielsen** [comment]
Reopen until we have investigated if there is a further issue with pyqt

**Comment by erikawa-e** [comment]
conda list

# Name                    Version                   Build  Channel
alabaster                 0.7.12                     py_0    conda-forge
argcomplete               1.12.3             pyhd8ed1ab_2    conda-forge
argh                      0.26.2          pyh9f0ad1d_1002    conda-forge
argon2-cffi               21.1.0           py37hcc03f2d_2    conda-forge
arrow                     1.2.1              pyhd8ed1ab_0    conda-forge
asteval                   0.9.25                   pypi_0    pypi
astroid                   2.9.0            py37h03978a9_0    conda-forge
async_generator           1.10                       py_0    conda-forge
atomicwrites              1.4.0              pyh9f0ad1d_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
autopep8                  1.6.0              pyhd8ed1ab_1    conda-forge
azure-core                1.21.1                   pypi_0    pypi
azure-identity            1.7.1                    pypi_0    pypi
babel                     2.9.1              pyh44b312d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
bcrypt                    3.2.0            py37hcc03f2d_2    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
black                     21.12b0            pyhd8ed1ab_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
broadbean                 0.10.0                   pypi_0    pypi
brotlipy                  0.7.0           py37hcc03f2d_1003    conda-forge
ca-certificates           2021.10.8            h5b45459_0    conda-forge
cachetools                4.2.4                    pypi_0    pypi
certifi                   2021.10.8        py37h03978a9_1    conda-forge
cffi                      1.15.0           py37hd8e9650_0    conda-forge
chardet                   4.0.0            py37h03978a9_2    conda-forge
charset-normalizer        2.0.9              pyhd8ed1ab_0    conda-forge
click                     8.0.3            py37h03978a9_1    conda-forge
cloudpickle               2.0.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
contextlib2               0.5.5                      py_2    conda-forge
cookiecutter              1.7.0                      py_0    conda-forge
cryptography              36.0.1           py37h65266a2_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
dataclasses               0.8                pyhc8e2a94_3    conda-forge
debugpy                   1.5.1            py37hf2a7229_0    conda-forge
decorator                 5.1.0              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
diff-match-patch          20200713           pyh9f0ad1d_0    conda-forge
docutils                  0.17.1           py37h03978a9_1    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
flake8                    4.0.1              pyhd8ed1ab_0    conda-forge
freetype                  2.10.4               h546665d_1    conda-forge
future                    0.18.2           py37h03978a9_4    conda-forge
google-api-core           2.3.2                    pypi_0    pypi
google-auth               2.3.3                    pypi_0    pypi
googleapis-common-protos  1.54.0                   pypi_0    pypi
h5netcdf                  0.11.0                   pypi_0    pypi
h5py                      2.10.0          nompi_py37he280515_106    conda-forge
hdf5                      1.10.6          nompi_h5268f04_1114    conda-forge
hypothesis                6.0.0              pyhd8ed1ab_0    conda-forge
icu                       64.2                 he025d50_1    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
imagesize                 1.3.0              pyhd8ed1ab_0    conda-forge
importlib-metadata        4.8.2            py37h03978a9_0    conda-forge
importlib_metadata        4.8.2                hd8ed1ab_0    conda-forge
importlib_resources       5.4.0              pyhd8ed1ab_0    conda-forge
inflection                0.5.1              pyh9f0ad1d_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
intel-openmp              2021.4.0          h57928b3_3556    conda-forge
intervaltree              3.0.2                      py_0    conda-forge
ipykernel                 6.6.0            py37h4038f58_0    conda-forge
ipython                   7.30.1           py37h03978a9_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
isort                     5.10.1             pyhd8ed1ab_0    conda-forge
jbig                      2.1               h8d14728_2003    conda-forge
jedi                      0.18.1           py37h03978a9_0    conda-forge
jellyfish                 0.8.9            py37hcc03f2d_2    conda-forge
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
jinja2-time               0.2.0                      py_2    conda-forge
jpeg                      9d                   h8ffe710_0    conda-forge
jsonschema                4.3.0              pyhd8ed1ab_0    conda-forge
jupyter                   1.0.0            py37h03978a9_7    conda-forge
jupyter_client            7.1.0              pyhd8ed1ab_0    conda-forge
jupyter_console           6.4.0              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.1            py37h03978a9_1    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
keyring                   23.4.0           py37h03978a9_0    conda-forge
kiwisolver                1.3.2            py37h8c56517_1    conda-forge
krb5                      1.19.2               h20d022d_3    conda-forge
lazy-object-proxy         1.7.1            py37hcc03f2d_0    conda-forge
lcms2                     2.12                 h2a16943_0    conda-forge
lerc                      3.0                  h0e60522_0    conda-forge
libblas                   3.9.0              12_win64_mkl    conda-forge
libcblas                  3.9.0              12_win64_mkl    conda-forge
libcurl                   7.80.0               h789b8ee_1    conda-forge
libdeflate                1.8                  h8ffe710_0    conda-forge
libiconv                  1.16                 he774522_0    conda-forge
liblapack                 3.9.0              12_win64_mkl    conda-forge
libpng                    1.6.37               h1d00b33_2    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libspatialindex           1.9.3                h39d44d4_4    conda-forge
libssh2                   1.10.0               h680486a_2    conda-forge
libtiff                   4.3.0                hd413186_2    conda-forge
libxml2                   2.9.12               hf5bbc77_1    conda-forge
libxslt                   1.1.33               h65864e5_3    conda-forge
libzlib                   1.2.11            h8ffe710_1013    conda-forge
lmfit                     1.0.3              pyhd8ed1ab_0    conda-forge
lxml                      4.7.1            py37hd07aab1_0    conda-forge
lz4-c                     1.9.3                h8ffe710_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.0.1            py37hcc03f2d_1    conda-forge
matplotlib                3.3.2            py37h03978a9_1    conda-forge
matplotlib-base           3.3.2            py37h3379fd5_1    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
mistune                   0.8.4           py37hcc03f2d_1005    conda-forge
mkl                       2021.4.0           h0e2418a_729    conda-forge
more-itertools            8.12.0             pyhd8ed1ab_0    conda-forge
msal                      1.16.0                   pypi_0    pypi
msal-extensions           0.3.1                    pypi_0    pypi
msys2-conda-epoch         20160418                      1    conda-forge
mypy_extensions           0.4.3            py37h03978a9_4    conda-forge
nbclient                  0.5.9              pyhd8ed1ab_0    conda-forge
nbconvert                 6.3.0            py37h03978a9_1    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
notebook                  6.4.6              pyha770c72_0    conda-forge
numpy                     1.19.2           py37he804e03_1    conda-forge
numpydoc                  1.1.0                      py_1    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
opencensus                0.8.0                    pypi_0    pypi
opencensus-context        0.1.2                    pypi_0    pypi
opencensus-ext-azure      1.1.0                    pypi_0    pypi
openjpeg                  2.4.0                hb211442_1    conda-forge
openssl                   1.1.1l               h8ffe710_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.3.5            py37h9386db6_0    conda-forge
pandoc                    2.16.2               h8ffe710_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
paramiko                  2.8.1              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pathspec                  0.9.0              pyhd8ed1ab_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.4.0            py37hd7d9ad0_0    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
platformdirs              2.3.0              pyhd8ed1ab_0    conda-forge
plottr                    0.8.1            py37hf2a7229_0    conda-forge
pluggy                    1.0.0            py37h03978a9_2    conda-forge
portalocker               2.3.2                    pypi_0    pypi
poyo                      0.5.0                      py_0    conda-forge
prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.24             pyha770c72_0    conda-forge
prompt_toolkit            3.0.24               hd8ed1ab_0    conda-forge
protobuf                  3.19.1                   pypi_0    pypi
psutil                    5.8.0            py37hcc03f2d_2    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycodestyle               2.8.0              pyhd8ed1ab_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydocstyle                6.1.1              pyhd8ed1ab_0    conda-forge
pyflakes                  2.4.0              pyhd8ed1ab_0    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyjwt                     2.3.0                    pypi_0    pypi
pylint                    2.12.2             pyhd8ed1ab_0    conda-forge
pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
pynacl                    1.4.0            py37ha54c9ec_3    conda-forge
pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.6              pyhd8ed1ab_0    conda-forge
pyqt                      5.9.2            py37h6538335_4    conda-forge
pyqt5                     5.15.6                   pypi_0    pypi
pyqt5-qt5                 5.15.2                   pypi_0    pypi
pyqt5-sip                 12.9.0                   pypi_0    pypi
pyqtgraph                 0.12.3                   pypi_0    pypi
pyreadline                2.1             py37h03978a9_1005    conda-forge
pyrsistent                0.18.0           py37hcc03f2d_0    conda-forge
pyside2                   5.15.2                   pypi_0    pypi
pysocks                   1.7.1            py37h03978a9_4    conda-forge
pytest                    6.2.5            py37h03978a9_1    conda-forge
pytest-runner             5.3.1              pyhd8ed1ab_0    conda-forge
python                    3.7.9           h7840368_100_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-lsp-black          1.0.1              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.3.3              pyhd8ed1ab_0    conda-forge
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyvisa                    1.11.1                   pypi_0    pypi
pywin32                   302              py37hcc03f2d_2    conda-forge
pywin32-ctypes            0.2.0           py37h03978a9_1004    conda-forge
pywinpty                  1.1.6            py37h7f67f24_0    conda-forge
pyyaml                    6.0              py37hcc03f2d_3    conda-forge
pyzmq                     22.3.0           py37hcce574b_1    conda-forge
qcodes                    0.30.1                   pypi_0    pypi
qdarkstyle                3.0.2              pyhd8ed1ab_0    conda-forge
qstylizer                 0.2.1              pyhd8ed1ab_0    conda-forge
qt                        5.9.7                h506e8af_3    conda-forge
qtawesome                 1.1.1              pyhd8ed1ab_0    conda-forge
qtconsole                 5.2.2              pyhd8ed1ab_0    conda-forge
qtpy                      1.11.3             pyhd8ed1ab_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_1    conda-forge
requirements-parser       0.2.0                      py_0    conda-forge
rope                      0.22.0             pyhd8ed1ab_0    conda-forge
rsa                       4.8                      pypi_0    pypi
rtree                     0.9.7            py37h13cc57e_3    conda-forge
ruamel-yaml               0.17.17                  pypi_0    pypi
ruamel-yaml-clib          0.2.6                    pypi_0    pypi
schema                    0.7.5              pyhd8ed1ab_0    conda-forge
scipy                     1.7.3            py37hb6553fb_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                59.6.0           py37h03978a9_0    conda-forge
setuptools-scm            6.3.2              pyhd8ed1ab_0    conda-forge
shiboken2                 5.15.2                   pypi_0    pypi
sip                       4.19.8          py37h6538335_1000    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
sphinx                    4.3.1              pyh6c4a22f_0    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_1    conda-forge
spyder                    5.2.1            py37h03978a9_0    conda-forge
spyder-kernels            2.2.0            py37h03978a9_0    conda-forge
sqlite                    3.37.0               h8ffe710_0    conda-forge
tabulate                  0.8.9              pyhd8ed1ab_0    conda-forge
tbb                       2021.4.0             h2d74725_1    conda-forge
terminado                 0.12.1           py37h03978a9_1    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
textdistance              4.2.2              pyhd8ed1ab_0    conda-forge
three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
tinycss2                  1.1.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.11               h8ffe710_1    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     1.2.2              pyhd8ed1ab_0    conda-forge
tornado                   6.1              py37hcc03f2d_2    conda-forge
tqdm                      4.62.3             pyhd8ed1ab_0    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
typed-ast                 1.5.1            py37hcc03f2d_0    conda-forge
typing-extensions         4.0.1                hd8ed1ab_0    conda-forge
typing_extensions         4.0.1              pyha770c72_0    conda-forge
ucrt                      10.0.20348.0         h57928b3_0    conda-forge
ujson                     4.2.0            py37hf2a7229_1    conda-forge
uncertainties             3.1.6              pyhd8ed1ab_0    conda-forge
urllib3                   1.26.7             pyhd8ed1ab_0    conda-forge
vc                        14.2                 hb210afc_5    conda-forge
vs2015_runtime            14.29.30037          h902a5da_5    conda-forge
watchdog                  2.1.6            py37h03978a9_1    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websockets                8.1              py37hcc03f2d_3    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
whichcraft                0.6.1                      py_0    conda-forge
widgetsnbextension        3.5.2            py37h03978a9_1    conda-forge
win_inet_pton             1.1.0            py37h03978a9_3    conda-forge
winpty                    0.4.3                         4    conda-forge
wrapt                     1.13.3           py37hcc03f2d_1    conda-forge
xarray                    0.20.2             pyhd8ed1ab_0    conda-forge
xz                        5.2.5                h62dcd97_1    conda-forge
yaml                      0.2.5                he774522_0    conda-forge
yapf                      0.31.0             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11            h8ffe710_1013    conda-forge
zstd                      1.5.0                h6255e5f_0    conda-forge

**Comment by jenshnielsen** [comment]
If you look carefully you can see that you have Pyqt installed with both conda and pip 

pyqt 5.9.2 py37h6538335_4 conda-forge
pyqt5 5.15.6 pypi_0 pypi
pyqt5-qt5 5.15.2 pypi_0 pypi
pyqt5-sip 12.9.0 pypi_0 pypi

this does not work and will result in errors like the one you see.

Both plottr and qcodes are available from conda forge so I suggest that you consider installing all packages from there

**Comment by erikawa-e** [comment]
Thanks.

As we discussed here (https://github.com/QCoDeS/Qcodes/issues/2824), I thought that it would be handy to have a cloned source.. I do not how to do this with conda forge... Is it possible with conda forge?

**Comment by jenshnielsen** [comment]
Unfortunatly not. In that case there is at the moment no other solution than to be careful and ensure that you do not mix pyqt packges from the 2 sources.

If you install plottr with `pip install -e .` it will install it without trying to install the qt binding. This should work if you have qt installed from conda. 

If you install it with `pip install -e .[pyqt]` then it will also insstall pyqt but then you need to be careful not to install pyqt from conda. See https://github.com/toolsforexperiments/plottr#installation

**Comment by erikawa-e** [comment]
Thanks for all your help.
Now I tried not use conda (after resetting the computer itself).

Then, I received the following error

> Exception: pyqtgraph requires Qt version >= 5.12  (your version is 5.9.6)

Do you have any suggestion?

**Comment by jenshnielsen** [comment]
The version of pyqt is too old. I think the version shipped by conda in the default channel may be very old but you can try to install a newer one using the conda forge channel

**Comment by erikawa-e** [comment]
Thanks.

Now with
"conda config --add channels conda-forge"
"conda config --set channel_priority strict"
"conda install 'pyqt>=5.12'",

I got the following errors;
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/noarch::argcomplete==1.12.3=pyhd3eb1b0_0
  - defaults/noarch::flake8==3.9.2=pyhd3eb1b0_0
  - defaults/noarch::importlib_metadata==4.8.2=hd3eb1b0_0
  - defaults/win-64::ipykernel==6.4.1=py37haa95532_1
  - defaults/noarch::ipywidgets==7.6.5=pyhd3eb1b0_1
  - defaults/noarch::jsonschema==3.2.0=pyhd3eb1b0_2
  - defaults/win-64::jupyter==1.0.0=py37_7
  - defaults/noarch::jupyter_console==6.4.0=pyhd3eb1b0_0
  - defaults/win-64::keyring==23.4.0=py37haa95532_0
  - defaults/noarch::nbclient==0.5.3=pyhd3eb1b0_0
  - defaults/win-64::nbconvert==6.1.0=py37haa95532_0
  - defaults/noarch::nbformat==5.1.3=pyhd3eb1b0_0
  - defaults/win-64::notebook==6.4.6=py37haa95532_0
  - defaults/win-64::pluggy==0.13.1=py37haa95532_0
  - defaults/noarch::pyls-spyder==0.4.0=pyhd3eb1b0_0
  - defaults/win-64::pytest==6.2.4=py37haa95532_2
  - defaults/noarch::pytest-runner==5.3.1=pyhd3eb1b0_0
  - defaults/noarch::python-lsp-black==1.0.0=pyhd3eb1b0_0
  - defaults/noarch::python-lsp-server==1.2.4=pyhd3eb1b0_0
  - defaults/noarch::qtconsole==5.1.1=pyhd3eb1b0_0
  - defaults/win-64::spyder==5.1.5=py37haa95532_1
  - defaults/win-64::spyder-kernels==2.1.3=py37haa95532_0
  - defaults/win-64::widgetsnbextension==3.5.1=py37_0

**Comment by jenshnielsen** [comment]
Yes using conda forge together with the default channel is not easy. I would suggest that you create a new env and install all dependencies from conda-forge

**Comment by erikawa-e** [comment]
Thanks.
I still wanted to stick to having a cloned source and I realized that I do not have any pyqt5 with 'pip list' for this installation. 

After 'pip install pyqt5', now "plottr-inspectr" finally works!!

But I got the following errors;
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
spyder 5.1.5 requires pyqtwebengine<5.13, which is not installed.
spyder 5.1.5 requires pyqt5<5.13, but you have pyqt5 5.15.6 which is incompatible.

So the spyder stopped working.

Then, after 'pip install pyqt5==5.12.3' and 'pip install pyqtwebengine==5.12.1',
finally both spyder and plottr works!!

Thanks so much for your patience.
